### PR TITLE
Add settings for initial limits

### LINKED ISSUE #71: Defaults for newly created sessions
A newly created session can be used for control data (WT_MAX_STREAMS, etc...) and datagrams, but it can't be used for reliable application data.  This is because the initial limits for streams is zero and the initial limits for stream data (both session-level and stream-level limits) is also zero.

We might consider setting baseline defaults for sessions.  These would likely need to be small because there would be no way to retract any value that was set in a default.  That is, if we set a stream limit of 10 (for any type of stream), then all implementations would be required to accept 10 streams.

If we go this way, we might use HTTP/2 stream limit defaults as a guide.  For stream data limits at least, we might allow 64k of data (the HTTP/2 default for [SETTINGS_INITIAL_WINDOW_SIZE](https://datatracker.ietf.org/doc/html/rfc9113#section-6.5.2-2.8.1)) relying on the same reasoning as HTTP/2 (which wasn't that thoroughly reasoned, to be honest, but whatever).  We cannot use that approach for stream limits as HTTP/2 sets an unlimited default, which doesn't work with the style of limiting we use.  Also, stream limits are independent of any HTTP/2 resources needed for tracking streams, so that would not apply.

Alternatively, we might use the current value of the HTTP/2 stream limits.  That is, the value of [SETTINGS_INITIAL_WINDOW_SIZE](https://datatracker.ietf.org/doc/html/rfc9113#section-6.5.2-2.8.1) might be input to a function that determines session- and stream-level limits.  This might also work if [SETTINGS_MAX_CONCURRENT_STREAMS](https://datatracker.ietf.org/doc/html/rfc9113#section-6.5.2-2.6.1) is set, but we might need a cap otherwise.

### LINKED ISSUE #72: Settings for newly created sessions
As an extension or alternative to #71, we might consider HTTP/2 settings that include initial values for WebTransport sessions.  Having non-zero initial values for stream and data limits would improve performance by allowing clients and servers to start sending reliable application data at the same time that they create a session.

The obvious drawback of this approach is that settings apply to connections, whereas the limits in question are resource-based.  That has two aspects: there might be multiple WebTransport resources on the same origin that have different applications and therefore different constraints; and, in deployments where there are intermediaries involved, this requires coordination across multiple connections in order to work effectively, especially if an intermediary serves multiple clients or servers and coalesces requests from more than one peer onto a single connection.  It is possible to present the lowest acceptable value across those supported by resources or remote peers, but establishing what that value is might be operationally challenging.

### PR DESCRIPTION
**Pull request by ekinnear**
Add settings for initial limits, with defaults of "0".

Fixes #72
Fixes #71

### CODE REVIEW

File: draft-ietf-webtrans-http2.md [hunk 1]
```diff
@@ -924,6 +932,113 @@ Specification:
 
 : This document
 
+The SETTINGS_WEBTRANSPORT_INITIAL_MAX_DATA parameter indicates the initial value
+for the session data limit, otherwise communicated by the WT_MAX_DATA capsule
+(see {{WT_MAX_DATA}}). The default value for the
+SETTINGS_WEBTRANSPORT_INITIAL_MAX_DATA parameter is "0", indicating that the
+endpoint will send a WT_MAX_DATA capsule within each session before its peer is
```
Comment (martinthomson) [inline draft-ietf-webtrans-http2.md:939]:
```suggestion
endpoint needs to send a WT_MAX_DATA capsule within each session before its peer is
```
...and the same for all future instances of the same.

File: draft-ietf-webtrans-http2.md [hunk 2]
```diff
@@ -924,6 +932,113 @@ Specification:
...
+Default:
+
+: 0
+
+Specification:
+
+: This document
+
+The SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAM_DATA parameter indicates the
+initial value for the stream data limits, otherwise communicated by the
```
Comment (martinthomson) [inline draft-ietf-webtrans-http2.md:962]: Don't we want to split this?  WT_MAX_STREAM_DATA applies to each stream individually, but you might want to have different bidirectional and unidirectional stream handling.

**Review by martinthomson — COMMENTED** [review]
Just a question about the bidi/uni split on stream data.  Otherwise, this looks solid.

File: draft-ietf-webtrans-http2.md (hunk 2 above)
Comment (ekinnear) [reply draft-ietf-webtrans-http2.md]: Good call, split these out in [5dec3cb](https://github.com/ietf-wg-webtrans/draft-ietf-webtrans-http2/pull/78/commits/5dec3cbe55823ef3f384a3e65ad2f6f985c0e3f6)

File: draft-ietf-webtrans-http2.md [hunk 3]
```diff
@@ -578,6 +581,11 @@ WT_MAX_STREAM_DATA capsules for flow control purposes and MUST generate and
 send appropriate flow control signals for their limits; see
 {{flow-control-intermediaries}}.
 
+Initial values for this limit for unidirectional and bidirectional streams MAY
+be communicated by sending non-zero values for
+SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAM_DATA_UNI and
+SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAM_DATA_BIDI.
```
Comment (martinthomson) [inline draft-ietf-webtrans-http2.md:587]:
```suggestion
SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAM_DATA_BIDI respectively.
```

**Review by martinthomson — APPROVED** [review]
Very good, thanks Eric.

**Comment by DavidSchinazi** [comment]
Chair: discussed in editor's meeting. @vasilvv mentioned that we need to discuss how to handle the case when the header and settings disagree. Simplest proposal is to use the max of the two, possibly in a separate section. Also mention that header should be higher than setting.

**Review by DavidSchinazi — APPROVED** [review]

File: draft-ietf-webtrans-http2.md [hunk 4]
```diff
@@ -261,13 +261,67 @@ too many sessions.
...
+#### Flow Control SETTINGS {#flow-control-settings}
+
+Initial flow control limits can be exchanged via HTTP/2 SETTINGS
+({{h2-settings}}) by providing non-zero values for
+
+* WT_MAX_DATA via SETTINGS_WEBTRANSPORT_INITIAL_MAX_DATA
+* WT_MAX_STREAM_DATA via SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAM_DATA_UNI and
+  SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAM_DATA_BIDI
+* WT_MAX_STREAMS via SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAMS_UNI and
+  SETTINGS_WEBTRANSPORT_INITIAL_MAX_STREAMS_BIDI
```
Comment (martinthomson) [inline draft-ietf-webtrans-http2.md:300]: Do you want to add forward references?

**Review by martinthomson — APPROVED** [review]

File: draft-ietf-webtrans-http2.md (hunk 4 above)
Comment (ekinnear) [reply draft-ietf-webtrans-http2.md:306]: Great idea, added in https://github.com/ietf-wg-webtrans/draft-ietf-webtrans-http2/pull/78/commits/13f622ff982d261277d9461bfcf03a8ab93a7617

**Review by vasilvv — APPROVED** [review]

**Pull request merged**